include/kernels/madwf_transfer.cuh

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include <color_spinor_field.h>
+#include <color_spinor_field_order.h>
+#include <shared_memory_cache_helper.cuh>
+#include <madwf_transfer.h>
+#include <madwf_ml.h>
+
+/**
+  @file This file contains the argument and kernel that applies fifth dimension transfer to a vector:
+      out_s = T_{st} * in_t,
+  where s and t are fifth dimension indices. T_{st} is a spin matrix so T has shape Ls*4-by-Ls*4.
+*/
+
+namespace quda
+{
+
+  namespace madwf_ml
+  {
+    constexpr int spin_dim = 4;
```

Comment:
I think we should use the variable conventions we use elsewhere of `nSpin`, `nColor`, etc, as opposed to `spin_dim`, etc. I don't necessarily have a good idea for `color_spin_dim` or `wm_dim`, though. What does `wm` stand for?

Reply:
I also think it's a bit weird they're floating around outside of a class, couldn't `nColor`, etc, be template parameters to the various classes like they are elsewhere in the code?

Reply:
`wm` is the dimension of a Wilson matrix. `sm` means the dimension of a spin matrix. Let me try to use `nSpin` and `nColor`, and also see if I can make things simpler without using `wm` or `sm`.

Reply:
About `nColor` being a template parameter: here we are only intending to have `nColor = 3`. The other option is to have a `nColor` template and add a `static_assert` to make sure it is `3`, just it seems redundant to me.

Reply:
https://github.com/lattice/quda/pull/1061/commits/322c1fea1cac8aa32988f4cee0a5e5b51c628344